### PR TITLE
Add tests for "project root" and "strip path" resolution

### PR DESCRIPTION
**Pull request by imjoehaines**
## Goal

This PR adds tests for the existing behaviour of the "project root" and "strip path" configuration options to support a PR extending them

## Tests

This PR only contains tests

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -140,7 +140,7 @@ matrix:
 
 before_install:
   - travis_retry composer require "laravel/framework:${LARAVEL_VERSION}" --no-update
-  - echo "memory_limit=2G" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
+  - echo "memory_limit=4G" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
```
Comment (imjoehaines) [inline .travis.yml:143]: This is from https://github.com/bugsnag/bugsnag-laravel/pull/394 to get the tests passing on Travis

File: tests/AbstractTestCase.php [hunk 2]
```diff
@@ -18,4 +18,21 @@ protected function getServiceProviderClass($app)
...
+     * @return mixed
+     */
+    protected function getProperty($object, $property)
+    {
+        $propertyAccessor = function ($property) {
+            return $this->{$property};
+        };
+
+        return $propertyAccessor->bindTo($object, $object)($property);
+    }
```
Comment (imjoehaines) [inline tests/AbstractTestCase.php:37]: I'd usually avoid things like this but it's kind of hard to tests these configuration options using the methods that exist in [the `Bugsnag\Configuration` class](https://github.com/bugsnag/bugsnag-php/blob/master/src/Configuration.php), but I'm happy to change the approach

File: tests/ServiceProviderTest.php [hunk 3]
```diff
@@ -37,6 +45,163 @@ public function testBugsnagLoggerIsInjectable()
...
+    }
+
+    public function projectRootAndStripPathProvider()
+    {
+        return [
+            // If both parameters are provided, the project root should be null
+            // and the strip path set to a regex matching the given string
+            // TODO this might be a bug — when a strip path value is configured,
+            //      we only set the project root path if one wasn't provided. If both
+            //      strip path _and_ project root are given, no project root is set
```
Comment (imjoehaines) [inline tests/ServiceProviderTest.php:148]: I think this is a bug, but it still needs investigation to see what the consequences are (relevant code here](https://github.com/bugsnag/bugsnag-laravel/blob/master/src/BugsnagServiceProvider.php#L341-L349))

File: tests/ServiceProviderTest.php [hunk 4]
```diff
@@ -37,6 +45,163 @@ public function testBugsnagLoggerIsInjectable()
...
+
+            // If only the strip path is provided, both values should be set to
+            // the same regex matching the given strip path string with "/app"
+            // appended
+            // TODO this might be a bug — when only strip path is configured,
+            //      we set the strip path and then immediately overwrite it with
+            //      a new path with "/app" appended by calling `setProjectRoot`.
+            //      If the intention is to to have "/example" be the strip path
+            //      and "/example/app" be the project root then we need to do these
+            //      calls in the opposite order
```
Comment (imjoehaines) [inline tests/ServiceProviderTest.php:183]: I think this is also a bug where both paths end up with `/app` appended as it looks like the intention is for only the project root to end in `/app` — [relevant code here](https://github.com/bugsnag/bugsnag-laravel/blob/master/src/BugsnagServiceProvider.php#L341-L349)
